### ISSUE TITLE
Likes: create a Like button Block

### ISSUE THREAD

**Issue by apeatling** [Automattic/jetpack#19659]
Implement a simple MVP server side rendering block for the WordPress.com like button:

<img width="258" alt="Screen Shot 2021-04-26 at 2 58 00 PM" src="https://user-images.githubusercontent.com/1464705/116156082-e102cc80-a69f-11eb-9a74-c22ed835fc09.png">

* Render the button server side in the editor
* Render the button server side in the front end
* Try and use all of the existing scripts that make this work

*****

Epic: #31920
PT: pdDOJh-2JP-p2
```[tasklist]
### Tasks
- [ ] https://github.com/Automattic/jetpack/pull/34340
```

**Comment by apeatling** [comment]
I'm positive I've seen some prior work on this that was abandoned. I cannot seem to find it again though, so please have a good search for this before starting the work.

**Comment by andrewserong** [comment]
Adding myself as an assignee, I'm really curious to see how we can make this work. Feel free to jump in too, if anyone else would like to!

**Comment by andrewserong** [comment]
We might not need to worry about this too much at the outset, but I wonder if we might need to make sure the distinction is clear for users between this particular Like feature and the Crowdsignal Vote block:

![image](https://user-images.githubusercontent.com/14988353/116174636-254ea500-a752-11eb-8498-7548308145bf.png)

**Comment by andrewserong** [comment]
The Crowdsignal Applause block is very similar too:

![image](https://user-images.githubusercontent.com/14988353/116200245-732bd300-a77b-11eb-99d6-09746f8c4332.png)

**Comment by jeherve** [comment]
Worth noting the work the +c3pop2 team is doing on Likes, to avoid issues with third-party cookies. It may be worth putting the Like block work on pause until this is figured out.

**Comment by andrewserong** [comment]
Thanks for the heads-up, Jeremy! I chatted with @apeatling and the objective at the moment is just for us to familiarise ourselves with the existing code, and see what kind of work would be involved to create a Like button block, as it's no longer as high priority as we thought.

From testing it out, the existing Like button behaviour that appends to the end of post content seems to be working okay in an FSE site, so it doesn't feel _too_ urgent to me to implement the Like button as a block, but I thought I'd have a quick hack around and see what might be involved, anyway.

I've had a go in https://github.com/Automattic/jetpack/pull/19676 — it's pretty hacky, but renders a static approximation of the Like button in the editor, and then provides a server-rendered approach for the Like button to render it wherever it's placed (and override the behaviour that appends to the end of a post). Because it uses the same server-rendering as the existing approach, it doesn't look like this will affect the third-party cookies work on Likes, however if the work is still in progress and we don't know yet what will need to be changed, then yes, it might be a good idea for us to park this for now.

But, on to other things I discovered along the way! I noticed while working on it, that if we go down this route, then the Share buttons will need the same treatment. Otherwise, we can place the Like button wherever we want, but the Share buttons will still be appended to the end of the post content. Here's an example screenshot:

![image](https://user-images.githubusercontent.com/14988353/116365164-527d7f00-a848-11eb-974f-bd89d47651ff.png)

So, I don't _think_ this will be too hard to implement in a more proper way, but it still could be a bit confusing to users to have these separate Like / Share blocks that are really just for placing this existing feature deliberately in a part of their template.

@apeatling I feel satisfied that when we'd like to implement this, it superficially doesn't seem _too_ complex to get it working. For now, is this enough info to go on? I'm happy to park this and the draft PR for when we'd like to circle back to it, and continue on with other tasks for this sprint. Otherwise, if you think it's higher priority, then I can keep chipping away.

**Comment by jeherve** [comment]
> I noticed while working on it, that if we go down this route, then the Share buttons will need the same treatment.

Indeed. That may be an even more important block, as more folks are using the Sharing feature than the Like one, by far.

**Comment by andrewserong** [comment]
I've written up a separate issue for the Sharing feature in https://github.com/Automattic/jetpack/issues/19694 but it might be good to tackle them at the same time. I'll see if I can squeeze in a little hacking around on that one this sprint, too, to get a sense of what's involved (I imagine a similar approach as to the Like button might work).

**Comment by apeatling** [comment]
> @apeatling I feel satisfied that when we'd like to implement this, it superficially doesn't seem too complex to get it working. For now, is this enough info to go on? I'm happy to park this and the draft PR for when we'd like to circle back to it, and continue on with other tasks for this sprint. Otherwise, if you think it's higher priority, then I can keep chipping away.

I think this is a lot of great info, and we can park it from here. Once it gets picked back up it will be useful as reference. Thanks for doing that. 👍

**Comment by jeherve** [comment]
Closing this one now that it's complete and live.
